packages/grid/x-data-grid-pro/src/hooks/features/columnReorder/useGridColumnReorder.tsx

```diff
@@ -317,8 +319,16 @@ export const useGridColumnReorder = (
       clearTimeout(removeDnDStylesTimeout.current);
       dragColNode.current = null;
 
+      const rect = apiRef.current.rootElementRef!.current!.getBoundingClientRect();
+      const hasDroppedOutside =
+        event.clientX < rect.left ||
+        event.clientX > rect.right ||
+        event.clientY < rect.top ||
+        event.clientY > rect.bottom;
```

Comment:
Unfortunately, this doesn't work in Safari.
For some reason, `clientX`/`clientY` values are wrong during the `dragend` event:

https://github.com/mui/mui-x/assets/13808724/613a2af3-fb5d-4953-b147-856ae28a6ebe

I've found a similar issue mentioned here:
> the clientX/screenX and clientY/screenY coordinates are simply wrong. I have no idea what they report. They don't even depend on where you touch.
https://horstmann.com/unblog/2018-12-16/index.html